### DISCUSSION TITLE
Drag canvas with right mouse button held

### DISCUSSION THREAD

**Discussion by Stefan-59** [zsviczian/obsidian-excalidraw-plugin#686] [Ideas]
I really LOVE Excalidraw / Obsidian, great thank you for all your effort! 🙏

One little enhanncment that I'd love to see:
When working with a drawing just with the mouse (moving around, zoom in and out, rearrange things, change some properties) currently I cannot move the canvas with the mouse without reaching to the keyboard and holding the space bar.

Would it be possible to implement that one can move the canvas by holding the right mouse button and move the mouse instead of requiring to use the keyboard?

**Comment by dcoales** [comment]
Just came to add the same request and then saw this one.  Would love to have this.

**Comment by sojusnik** [comment]
> I really LOVE Excalidraw / Obsidian, great thank you for all your effort! pray
> 
> One little enhanncment that I'd love to see: When working with a drawing just with the mouse (moving around, zoom in and out, rearrange things, change some properties) currently I cannot move the canvas with the mouse without reaching to the keyboard and holding the space bar.
> 
> Would it be possible to implement that one can move the canvas by holding the right mouse button and move the mouse instead of requiring to use the keyboard?

What a great idea! Did you think about proposing this upstream, in Excalidraw itself?

**Comment by FederaikLudikon** [comment]
Same for me, I have a wacom touch display tablet with my mac (by the way excalidraw is wonderful with it) and using the 'two fingers drag' I get the weird reverse movement. It would be amazing to have a setting to swap directions of panning and I'm sure it's a quick fix. Pretty pleeease 🤣
